webviz_subsurface/_components/parameter_filter.py

```diff
@@ -164,111 +166,399 @@ def _initial_store(self) -> Dict[str, List]:
+            Input({"id": self._uuid, "type": "slider", "name": ALL, "r": ALL}, "value"),
+            Input({"id": self._uuid, "type": "select", "name": ALL, "r": ALL}, "value"),
+            State({"id": self._uuid, "type": "slider", "name": ALL, "r": ALL}, "id"),
+            State({"id": self._uuid, "type": "select", "name": ALL, "r": ALL}, "id"),
+            State({"id": self._uuid, "type": "value-store"}, "data"),
+            State({"id": self._uuid, "type": "apply-button"}, "style"),
+        )
+        def _update_apply_button_style(
+            _n_clicks: int,
+            sliders: List[list],
+            selects: List[list],
+            slider_id: List[dict],
+            select_id: List[dict],
+            values_stored: Dict[str, list],
+            style: dict,
+        ) -> dict:
+
+            ctx = callback_context.triggered[0]["prop_id"]
+            values = self._get_values(sliders, selects, slider_id, select_id)
+            modified_values = any(
+                values[col] != values_stored.get(col, values[col]) for col in values
+            )
+
+            if modified_values and not "apply-button" in ctx:
+                style["background-color"] = "#7393B3"
+                style["color"] = "white"
+            else:
+                style["background-color"] = "white"
+                style["color"] = "#555"
+            return style
+
+        @callback(
+            Output({"id": self._uuid, "type": "ensemble-selector"}, "value"),
+            Output({"id": self._uuid, "type": "ensemble-selector"}, "options"),
+            Input({"id": self._uuid, "type": "ensemble-update"}, "data"),
+            prevent_initial_call=True,
+        )
+        def _update_ensembles_from_outside(ensembles: list) -> tuple:
+            """Update ensemble in parameter filter"""
+            return (
+                ensembles,
+                [
+                    {
+                        "label": val,
+                        "value": val,
+                        "disabled": val not in ensembles or len(ensembles) == 1,
+                    }
+                    for val in self._ensembles
+                ],
+            )
+
+
+def make_range_slider(
+    value: list, min_max: list, step: float, name: str, uuid: str
+) -> html.Div:
     return wcc.RangeSlider(
         label=name,
-        id={"id": uuid, "type": "range-slider", "name": name},
-        min=values.min(),
-        max=values.max(),
-        step=calculate_slider_step(
-            min_value=values.min(),
-            max_value=values.max(),
-            steps=len(list(values.unique())) - 1,
-        ),
-        value=[values.min(), values.max()],
-        marks={
-            str(values.min()): {"label": f"{values.min():.2f}"},
-            str(values.max()): {"label": f"{values.max():.2f}"},
+        id={
+            "id": uuid,
+            "type": "slider",
+            "name": name,
+            "r": str(random.random()),  # needed to trigger update (bug)
```

Comment:
try `uuid4` instead for the security issue

Reply:
good suggestion, I will 👍